### PR TITLE
Refactor Docker.linux.testrunner to use prebuilt image with Docker and PowerShell

### LINKED ISSUE #474: Rename PowerShell to pwsh
PowerShell is now **pwsh** as per https://github.com/PowerShell/PowerShell/releases/tag/v6.0.0-beta.9
>Starting from this release, the PowerShell Core executable is renamed to be pwsh

PowerShell version `v6.0.0-beta.6` is used in this repo.  For example at (https://github.com/dotnet/dotnet-docker-nightly/blob/master/test/Dockerfile.linux.testrunner#L29). If we were to upgrade to `v6.0.0-beta.9` or newer, then make sure to use `pwsh` in place of of `PowerShell`.

### LINKED ISSUE #521: [Test] Consider pushing Dockerfile.linux.testrunner image to Docker registry.
Currently the Dockerfile.linux.testrunner Dockerfile is built on every Linux build leg in the official builds and CI.  This Dockerfile is taking a significant amount of time to build.  Approximately 1/3 of the build time.  The Docker and Powershell parts of this Dockerfile could be built once and pushed to a Docker registry e.g. (https://hub.docker.com/r/microsoft/dotnet-buildtools-prereqs) and simply pulled in the build legs.  This change could be flown to the other Docker repos that have this same testrunner Dockerfile.

### PR DESCRIPTION
**Pull request by MichaelSimons**
Fixes #521 
Fixes #474

### CODE REVIEW

**Comment by MichaelSimons** [comment]
@dotnet-bot test ci please

**Comment by MichaelSimons** [comment]
@dotnet-bot test ci please

**Comment by MichaelSimons** [comment]
@dotnet-bot test Debian - 1. Dockerfiles please
@dotnet-bot test Debian - 2. Dockerfiles please

**Comment by MichaelSimons** [comment]
@dotnet-bot test ci please

**Comment by ravimeda** [comment]
Thanks very much for doing this, Michael!

File: test/Dockerfile.linux.testrunner [hunk 1]
```diff
@@ -1,34 +1,7 @@
 # Use this Dockerfile to create a Linux testrunner image
-#     docker build -t testrunner -f ./test/Dockerfile.linux.testrunner .
-#     docker run --rm -v /var/run/docker.sock:/var/run/docker.sock testrunner powershell -File build-and-test.ps1 -UseImageCache
+# Usage: docker run --rm -v /var/run/docker.sock:/var/run/docker.sock testrunner powershell -File build-and-test.ps1 <params....>
```
Comment (ravimeda) [inline test/Dockerfile.linux.testrunner]: Nit - replace `powershell` as `pwsh`

Comment (MichaelSimons) [reply test/Dockerfile.linux.testrunner]: Fixed

**Pull request merged**